### DISCUSSION TITLE
Not working on certain website. Downloading issues

### DISCUSSION THREAD

**Discussion by dwu105** [aclap-dev/video-downloadhelper#410] [Help and Q&A]
### Your question

I was trying to download videos from JFF, the adult site. It could detect the videos I was playing, but it just couldn't download properly. 

When it was working fine, it downloads the video from my Chrome's downloader. Then all of a sudden, it downloads from its internal downloader, which I think is the companion downloader, and that's when the errors occur. At first, it would stop the downloading process at 1% and appear as a side download error. I've tried it for a while, now it would download the video. Just I thought it was working fine in the companion downloader, but the downloaded video was corrupted. 

I tried to go to the settings and use the browser as the downloader, but it won't change its behavior. 

### Addon & CoApp version and details

version: 8.2.2.9
channel: stable
build date: 2024-03-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: accepted
key (only 16 first characters): 1f03ecb96f574cdb
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "D:\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "i686",
    "node": "10"
  },
  "home": "C:\\Users\\Administrator",
  "converterBinary": "D:\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by Mattmjs** [comment]
>  the downloaded video was corrupted.

Sounds to me like this site uses DRM protection. You can verify this but it only works in Firefox. If you see the icon in the image
then nothing can be done for this site.

![DRM icon](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/f48a1898-9a58-443f-89e9-a552d3f2dbc2)

**Reply by dwu105** [reply]
No, it does not have that icon on my Chrome 
![Screenshot 2024-04-20 143134](https://github.com/aclap-dev/video-downloadhelper/assets/166884497/80726f72-3e96-4456-b6f3-00787372d7a0)

**Reply by dwu105** [reply]
I tried firefox. Same problem I have with Chrome.

**Reply by Mattmjs** [reply]
I told you, it only works in Firefox. You won't see that icon on Chrome. Do you see it in Firefox ?

**Reply by mcajina74** [reply]
I have the same issue with JFF website. I'm using Firefox on Windows. I don't see that "chain" DRM icon in the URL field.

The video looks like this when viewed on VLC:

![CorruptedVideo](https://github.com/aclap-dev/video-downloadhelper/assets/11951639/130d5322-8e6a-4d31-8c98-f967fac800eb)

**Reply by Mattmjs** [reply]
I don't know why you never saw the icon, it should be there whenever the DRM setting is enabled.
But your image is an example of DRM protection, and there is no way around it.

**Reply by GrampaWildWilly** [reply]
Right.  Let's have the URL of the web page from which you downloaded that scrambled video.  Don't do any analysis.  Just visit the web page where the video is hosted, then post that URL here.

**Reply by mcajina74** [reply]
It's really even just from your "home" page after you log in. It just shows you a feed from your subscriptions:
https://justfor.fans/home

That little chain icon doesn't appear in the URL field.

I have noticed that some videos (usually previews) have a different interface when you click play. It's the "older" basic white on black background interface. These videos are downloadable by VDF. However, some of the previews and the full videos, have a different interface with blue interface elements. These are the ones that download corrupted.

**Reply by GrampaWildWilly** [reply]
We're not going to make any progress here unless we can get on some web pages within the site.  But it requires a login & I am not about to register for this site just to diagnose this problem.  I will say that it is not unusual for sites to put some of their content behind DRM protection but not other content.  Perhaps some content doesn't need to be protected.  Perhaps they are rolling out DRM & the content that is not protected now might be next week.  Perhaps perhaps perhaps.  As outsiders, we can't know the whole story.  I say just be happy with content VDH does download correctly & give up on the rest.
